alert_rules.py

```
@@ -0,0 +1,135 @@
+from datetime import datetime
+
+
+def get_season_from_month(month: int) -> str:
+    if month in [6, 7, 8, 9]:
+        return "kharif"
+    elif month in [11, 12, 1, 2]:
+        return "rabi"
+    return "zaid"
+
+def generate_alerts(
+    crop: str = None,
```

Ruff (0.15.11)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)